src/main/java/net/minecraftforge/fml/common/registry/FMLControlledNamespacedRegistry.java

```diff
@@ -768,7 +768,7 @@ public void loadIds(Map<ResourceLocation, Integer> ids, Map<ResourceLocation, In
             if (currId == -1)
             {
                 FMLLog.info("Found a missing id from the world %s", itemName);
-                missingIds.put(entry.getKey(), newId);
+                missingIds.put(itemName, newId);
```

Comment:
Not sure why this is changed, but after looking its just a microoptimization so meh.